src/RatcliffDDM.jl

```diff
@@ -1,371 +1,662 @@
+    S = S - y  # the last term should be f(b) and not 2*f(b) so we subtract y
+    S = S / (ub_t - lb_t)
+
+    return ht * S / 3
+
+end
+
+logpdf(d::RatcliffDDM, choice, rt; ϵ::Real = 1.0e-12) = log(pdf(d, choice, rt; ϵ))
+
+function logpdf(d::RatcliffDDM, data::T) where {T<:NamedTuple}
+    return sum(logpdf.(d, data...))
+end
+
+function logpdf(dist::RatcliffDDM, data::Array{<:Tuple,1})
+    LL = 0.0
+    for d in data
+        LL += logpdf(dist, d...)
+    end
+    return LL
+end
+
+logpdf(d::RatcliffDDM, data::Tuple) = logpdf(d, data...)
+
+########################################################################################################################################################################
+# Calculate Drift-diffusion Probability Density
+#
+# compute the densities g- and g+ of the first exit time. `cdf` implement A1 to A4 equations in Voss, Rothermund, and Voss (2004). These equations calculate Ratcliff's
+# drift-diffusion model (1978). This source codes are derived from Henrik Singmann's Density.h (rtdists) &  Voss & Voss's density.c (fast-dm).
+#
+#  * ------------------------------------------------------------------------
+#  * A verbatim copy of Jochen Voss & Andreas Voss's copyright.
+#  * ------------------------------------------------------------------------
+#  * Copyright (C) 2012  Andreas Voss, Jochen Voss.
+#  *
+#  * This program is free software; you can redistribute it and/or
+#  * modify it under the terms of the GNU General Public License as
+#  * published by the Free Software Foundation; either version 2 of the
+#  * License, or (at your option) any later version.
+#  *
+#  * This program is distributed in the hope that it will be useful, but
+#  * WITHOUT ANY WARRANTY; without even the implied warranty of
+#  * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+#  * General Public License for more details.
+#  *
+#  * You should have received a copy of the GNU General Public License
+#  * along with this program; if not, write to the Free Software
+#  * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+#  * 02110-1301 USA.
+#
+#  # References
+#
+# - Singmann H, Brown S, Gretton M, Heathcote A (2022). _rtdists: Response Time Distributions_. Rpackage version 0.11-5, <https://CRAN.R-project.org/package=rtdists>.
+# - Voss, A., Rothermund, K., & Voss, J. (2004). Interpreting the parameters of the diffusion model: An empirical validation. *Memory and Cognition, 32(7)*, 1206-1220.
+# - Ratcliff, R. (1978). A theory of memory retrieval. *Psychology Review, 85(2)*, 59-108.
+# - Voss, A., Voss, J., & Lerche, V. (2015). Assessing cognitive processes with diffusion model analyses: A tutorial based on fast-dm-30. *Frontiers in Psychology, 6*, Article 336. https://doi.org/10.3389/fpsyg.2015.00336
+#
+################################################################################################################################################################
+
+TUNE_PDE_DT_MIN = 1e-6
```

Comment:
In 99.9% of cases, global variables are not a good idea in any language, but they are particularly bad for performance in Julia. I think there are two options:

1. If you don't think these variables will be changed, you can define them in the functions where they are used. It looks like they are only used in one function each. So this might be a viable solution.

2. If you think people will change the values, you could define a struct containing the constants and add it to the model struct. The constants could be initialized it in the model constructor based on the keyword arguments with default values .